src/dualscreen.tsx

```diff
@@ -294,27 +289,23 @@ export class DualScreen extends KalturaPlayer.core.BasePlugin {
     );
   };
 
-  // private _getSecondaryMedia() {
-  //   const headers: Map<string, string> = new Map();
-  //   headers.set('Content-Type', 'application/json');
-  //   const request = new RequestBuilder(headers);
-  //   request.service = 'baseEntry';
-  //   request.action = 'list';
-  //   request.method = 'POST';
-  //   request.tag = 'list';
-  //   request.params = {
-  //     filter: {objectType: 'KalturaBaseEntryFilter', parentEntryIdEqual: this._player.getMediaInfo().entryId},
-  //     responseProfile: {
-  //       type: ResponseProfileType.INCLUDE_FIELDS,
-  //       fields: 'id'
-  //     }
-  //   };
-  //   this._player.provider.getCustomData({requests: [request]}).then((response: any) => {
-  //     this._secondaryKalturaPlayer.loadMedia({entryId: response[0].data.objects[0].id});
-  //     this._setMode();
-  //     this._videoSyncManager = new VideoSyncManager(this.eventManager, this.player, this._secondaryKalturaPlayer, this.logger);
-  //   });
-  // }
+  private _getSecondaryMedia() {
+    this._player.provider
+      .doRequest([{loader: SecondaryMediaLoader, params: {parentEntryId: this._player.getMediaInfo().entryId}}])
+      .then((data: Map<string, any>) => {
+        if (data && data.has(SecondaryMediaLoader.id)) {
+          const secondaryMediaLoader = data.get(SecondaryMediaLoader.id);
+          if (secondaryMediaLoader && secondaryMediaLoader.response) {
+            this._secondaryKalturaPlayer.loadMedia({entryId: secondaryMediaLoader.response[0].data.objects[0].id});
```

Comment:
we should avoid accessing nested objects like this, and in general we should parse the response objects and we have the response objects for this in the core provider library.

Reply:
anyway move this parsing to the loader's response setter (see media-entry-loader for examle)

Reply:
I agree, also consider using the base media list object from provider, again, if possible, cause maybe it is also coupled today to the provider use case only

Reply:
Just handled the response in the Setter